prow/cmd/deck/main.go

```diff
@@ -1274,8 +1286,24 @@ func canTriggerJob(user string, pj prowapi.ProwJob, cfg *prowapi.RerunAuthConfig
 	}
 	if cli == nil {
 		logrus.Warning("No GitHub token was provided, so we cannot retrieve GitHub teams")
+		return false, nil
+	}
+
+	// If the job is a presubmit and has an associated PR, do the same checks as for /test
```

Comment:
I know this may not be idiomatic Go but it might be nicer to have 

```
if (thing is presubmit and has a PR) {
    return result from trigger logic
}

return false
```